packages/core/src/modules/sd-jwt-vc/credential-status/token-status-list/TokenStatusList.ts

```diff
@@ -0,0 +1,237 @@
+    return statusList.getStatus(index)
+  }
+
+  async getStatusList(agentContext: AgentContext, uri: string): Promise<string> {
+    const registry = this.findRegistry({ uri })
+    if (!registry) {
+      throw new TokenStatusListError(`No token status list registry registered for uri ${uri}`)
+    }
+
+    return await registry.retrieve(agentContext, uri)
+  }
+
+  getStatusListFetcher(agentContext: AgentContext) {
+    return async (uri: string) => {
+      return await this.getStatusList(agentContext, uri)
+    }
+  }
+
+  async publishStatusList(
+    agentContext: AgentContext,
+    issuer: SdJwtVcIssuer,
+    jwt: string,
+    options: PublishTokenStatusListOptions
+  ): Promise<string> {
+    // verify jwt
+    const verified = await this.verifyStatusList(agentContext, issuer, jwt)
+    if (!verified.isValid) {
+      throw new TokenStatusListError('Invalid Jwt in the provided statusListUri')
+    }
+
+    const registry = this.findRegistry({ uri: options.uri, issuer })
+    if (!registry) {
+      throw new TokenStatusListError(`No token status list registry registered for issuer ${issuer}`)
+    }
+    return await registry.publish(agentContext, issuer, jwt, options)
+  }
+
+  private findRegistry({ uri, issuer }: { uri?: string; issuer?: SdJwtVcIssuer }): TokenStatusListRegistry | null {
+    let method: string
+
+    if (uri) {
+      if (isDid(uri)) {
+        method = parseDid(uri).method
+      } else if (isURL(uri)) {
+        method = 'http'
+      } else {
+        throw new TokenStatusListError('Status List Uri is not supported')
+      }
+    } else if (issuer && issuer.method === 'did') {
+      method = parseDid(issuer.didUrl).method
+    } else {
+      throw new TokenStatusListError('Status List Uri is not provided')
+    }
+
+    return this.sdJwtVcModuleConfig.registries.find((r) => r.supportedMethods.includes(method)) ?? null
+  }
+
+  private async verifyStatusList(agentContext: AgentContext, issuer: SdJwtVcIssuer, jwt: string) {
+    const jwsService = agentContext.dependencyManager.resolve(JwsService)
+    return await jwsService.verifyJws(agentContext, {
+      jws: jwt,
+      resolveJwsSigner:
+        issuer.method === 'did'
+          ? async ({ protectedHeader: { alg, kid } }) => {
+              if (!kid || typeof kid !== 'string') throw new CredoError('Missing kid in protected header.')
+
+              const { did } = parseDid(issuer.didUrl)
+              const didUrl = `${did}${kid}`
+              const didsApi = agentContext.dependencyManager.resolve(DidsApi)
+              const didDocument = await didsApi.resolveDidDocument(did)
+              const verificationMethod = didDocument.dereferenceKey(didUrl)
+              const publicJwk = getPublicJwkFromVerificationMethod(verificationMethod)
+
+              return {
+                alg,
+                method: issuer.method,
+                didUrl,
+                jwk: publicJwk,
+              }
+            }
+          : undefined,
```

Comment:
If the jwt is now signed with something else than `did` we don't check if it's actually signed by the SdJwtVcIssuer.

I think we should use `jwsSigner` instead and provide it beforehand.